scripts/calc_from_args.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+import logging
+import os
+import sys
+from argparse import ArgumentParser
+from datetime import datetime as dt
+
+from genologics.config import BASEURI, PASSWORD, USERNAME
+from genologics.entities import Process
+from genologics.lims import Lims
+
+from calc_from_args_utils import calculation_methods
+from scilifelab_epps.epp import upload_file
+
+DESC = """UDF-agnostic script to perform calculations across all artifacts of a step.
+
+The script is written with the intention of reusability,
+so the script arguments specify which UDFs to use and
+from where their values should be fetched.
+"""
+
+TIMESTAMP: str = dt.now().strftime("%y%m%d_%H%M%S")
+SCRIPT_NAME: str = os.path.basename(__file__).split(".")[0]
+
+
+def parse_udf_arg(arg_string: str) -> dict:
+    """Parse UDF argument string into a dictionary.
+
+    Example:
+
+        the argument
+
+            --vol_in udf='Volume (ul)',source='input',recursive=False
+
+        will assign
+
+            args.vol_in = {
+                'udf': 'Volume (ul)',
+                'source': 'input',
+                'recursive': False
+            }
+
+    The keys "source" and "recursive" have default values.
```

Comment:
Not really default values but rather that they need to confine to a strict set of choices? Maybe document here what the choices are?

Reply:
The defaults are applied [here](https://github.com/SciLifeLab/scilifelab_epps/pull/321/files/8eabf2372d2a6d4a9a2f4a8d25e3adbb8aaa852a#diff-9599aa3feb4778835f434a02ab919e1da58b069c9e5f123159f0720a7563e9acR64)

Reply:
```suggestion
    The keys "source" and "recursive" have default values "output" and False respectively.
```